packages/astro/src/i18n/index.ts

```diff
@@ -279,6 +279,28 @@ export function toPaths(locales: Locales): string[] {
 		}
 	});
 }
+/**
+ * Returns the pathname with a given locale either removed
+ * or optionally replaced. Does not touch parts of the
+ * pathname that look like a locale but aren’t
+ * @param pathname 
+ * @param localeToRemove 
+ * @param replacement 
+ */
+export function removeOrReplaceLocaleFromPathname(locale: string, pathname: string, replacement?: string) {
+	// We want to remove the default locale only when it is followed
+	// by `/` or nothing, not when it is part of a URL fragment, eg. 
+	// `/en/crypto/enigma` should not become 
+	// `/cryptoigma` if the default locale of `en` is stripped
+	const regex = new RegExp(`/${locale}(/|$)`)
```

Comment:
This is a costly operation, because we create a new regex every single time, without "caching" regexes that we already use. Can we achieve the same result with simple string operations?

Reply:
Wasn’t quite as simple, the fallback change added last week also needs a regex for at least one other locale, but either way, the regexes are now cached.